application/android/app/src/main/java/com/group12/beabee/network/ServiceAPI.java

```diff
@@ -127,4 +130,42 @@
 
     @POST("entities/question")
     Call<BasicResponse> createQuestion(@Body Question question);
+
+
+
+    //GROUP GOAL LINKS:
+    @PUT("groupgoals")
+    Call<BasicResponse> updateGG(@Body GroupGoalDTO userDTO);
+
+    @GET("groupgoals/{groupgoal_id}")
+    Call<GroupGoalDTO> getGGDetail(@Path("groupgoal_id") int goalId);
+
+    @DELETE("groupgoals/{groupgoal_id}")
+    Call<BasicResponse> deleteGG(@Path("groupgoal_id") int goalId);
+
+    @PUT("groupgoals/{groupgoal_id}/members")
+    Call<BasicResponse> addMember(@Body GroupGoalDTO goalDTO, UserDTO userDTO);
+
+    @POST("groupgoals/{groupgoal_id}/token")
+    Call<BasicResponse> createToken(@Path("groupgoal_id") int goalId);
+
+    @POST("groupgoals/{user_id}")
+    Call<BasicResponse> createGG(@Path("user_id") int user_id, @Body GroupGoalDTO ggDTO);
```

Comment:
Here on the body instead of `GroupGoalDTO` we can use `Goal` object from `Models/Requests`. Since DTO object contains the same fields and more it works fine but we are sending unnecessary information to the server and it might cause a problem in the future